### PR TITLE
disable unavailable heroes in selection

### LINKED ISSUE #2705: In selecting heroes from portraits unavailable heroes should be disabled
**Describe your proposal**
When using the new hero portraits selector,  you can select a hero that is picked by other player. (nothing happens, but its shown in the list and you can click it)

Ideally those heroes should not be available for selection, just like in the old left right arrows selection controls.

In the picture Tan selected Sir Mulich, but he is still available for Blue.
image

Proposition... show them greyed out? or with a red X? or if too complicated... not at all?

### PR DESCRIPTION
**Pull request by Laserlicht**
fixes #2705

@krs0 Can you test it?

grafik

### CODE REVIEW

**Comment by dydzio0614** [comment]
whoever tests this should include this test case: red - select town -> blue, select town and hero -> red: check if hero selected by blue is blocked

**Comment by dydzio0614** [comment]
BTW maybe reuse existing code for available heroes from old way of selecting them?

**Comment by krs0** [comment]
Looks fine. I have tested with a map that had Valeska banned, Orin placed on map and Sir Mulich in prison.
They did not appear
Heroes for other players appear correctly greyed out. (Christian)
image

**Comment by Laserlicht** [comment]
> BTW maybe reuse existing code for available heroes from old way of selecting them?

Client side do not have code for this. On the old way this is handled by server.

File: client/lobby/OptionsTab.h [hunk 1]
```diff
@@ -114,7 +114,8 @@ class OptionsTab : public InterfaceObjectConfigurable
 
 		std::vector<FactionID> factions;
 		std::vector<HeroTypeID> heroes;
-
+		std::vector<HeroTypeID> unusableHeroes;
```
Comment (Nordsoft91) [inline client/lobby/OptionsTab.h:117]: consider std::set, you have then:
`unusableHeroes.insert(player.second.hero)` and `if(unusableHeroes.count(elem)`

**Pull request merged**